consensus/src/dag/reconstruction/mod.rs

```diff
@@ -76,6 +77,23 @@ impl<U: Unit> UnitWithParents for ReconstructedUnit<U> {
     }
 }
 
+impl<D: Data, H: Hasher, K: MultiKeychain> From<ReconstructedUnit<Signed<FullUnit<H, D>, K>>>
+    for OrderedUnit<D, H>
+{
+    fn from(unit: ReconstructedUnit<Signed<FullUnit<H, D>, K>>) -> Self {
+        let parents = unit.parents;
+        let hash = unit.unit.hash();
```

Comment:
This and above could be accessed through methods (of `UnitWithParents` iirc) instead of fields.

Reply:
You can also get the creator and round this way btw.

Reply:
Actually I would prefer to do that using fields directly instead. I would like to treat it as a low-level constructor that knows its own internals instead of using generic traits. This way it fits better to implement `Into` directly (I know internals of `ReconstructedUnit<..>` and want to create `OrderedUnit`...but clippy complains about it :| ). Using generic traits here suggests that I should rather have something like `From<U: Unit>` instead, but this way I can't access `Data`, etc. That `.clone()` also smells a bit. Anyway, I've changed it.

Reply:
changed